client/state/sites/middleware.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * Internal dependencies
+ */
+import sitesList from 'lib/sites-list';
+import { receiveSites } from 'state/sites/actions';
+
+export default ( { dispatch } ) => {
+	const sites = sitesList();
+	sites.on( 'change', () => {
+		dispatch( receiveSites( sites.get() ) );
```

Comment:
Could we have issues with this `sites.get()` not being pure? For instance, `getRawSite()` would be selecting an already computed site object, and the operations we do on `getSite` could trip over expecting some of these fields to be empty?